pepper-apis/src/main/java/org/broadinstitute/ddp/Housekeeping.java

```diff
@@ -367,8 +367,15 @@ public static void main(String[] args) {
                                         try {
                                             message = messageBuilder.createMessage(ddpMessageId, pendingEvent, apisHandle);
                                         } catch (MessageBuilderException e) {
-                                            LOG.error("Could not create message for queued event "
-                                                    + pendingEvent.getQueuedEventId(), e);
+                                            if (e.shouldDeleteEvent()) {
+                                                LOG.warn("Unable to create message for event with queued_event_id={}, proceeding to delete",
+                                                        pendingEvent.getQueuedEventId(), e);
+                                                queuedEventDao.deleteAllByQueuedEventId(pendingEvent.getQueuedEventId());
+                                                return; // Exit out of transaction wrapper and move on to next event.
```

Comment:
Design-wise does not look right to me to have exception itself dictate to code catching it what to do (shouldDeleteEvent). A suggestion: maybe there is an "AddressNotAvailableException" (maybe subclass of MessageBuilder) and then the handler, based on the Study configuration setting you created in the PR, can decide on whether to create the event or not.

Reply:
Good point! I have moved the decision higher up in the caller now.